syntax = "proto3";

package video;

service VideoService {
    rpc ServerActive (Empty) returns (StatusResponse);
    rpc Upload (stream UploadRequest) returns (UploadResponse);
    rpc DeleteFile (DeleteRequest) returns (StatusResponse);
}

message Empty {}

message StatusResponse {
    string status = 1;
    string message = 2;
}

message UploadRequest {
    bytes video = 1;
    bytes audio = 2;
}

message UploadResponse {
    string video_url = 1;
    string audio_url = 2;
    string result_url = 3;
    string thumbnail_url = 4;
}

message DeleteRequest {
    string key = 1;
}
